### DISCUSSION TITLE
How can I diagnose startup time for my job between the Kubernetes run worker job created and the Started execution of run steps?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18190] [Q&A] [answered]
Hi all!

I am experience a lot of overhead (2min 20s) using `K8sRunLauncher` when starting the K8s job which will host a given run.

The overhead is highlighted in the picture. It happens between the *Kubernetes run worker job created* and the *Started execution of run for '&lt;job_name&gt;'*.

I believe it is not an issue in my K8s cluster since the job is created in the cluster almost instantaneously.

Can it be related to the time to load all the definitions in my user-deployed code?
How can I further investigate it?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700579906434389?thread_ts=1700579906.434389&cid=C01U954MEER).*

**Comment by gustavo-delfosim** [comment]
I am the original author of this question.
Posting the screenshot for additional information:
![image](https://github.com/dagster-io/dagster/assets/79655614/c1789f22-cbee-473f-821e-32b3a1376cf8)

**Comment by gibsondan** [comment, marked as answer]
Hi @gustavo-delfosim - i think this is indeed likely the time it takes to import your Python module with your definitions in it, as that's the main work that's happening during this time period.

Using py-spy is one tool that can help here: https://github.com/dagster-io/dagster/discussions/14771

Splitting things into more code locations is another option, as the run will only need to load the code location in which its ops or assets are in.
